torch/_inductor/scheduler.py

```diff
@@ -1649,54 +1708,60 @@ def add_user(
             else:
                 node_mode = None
 
-            # a node will mutate either 0 or 1 buffers
-            assert len(node.get_mutations()) <= 1
-            for alt_name in node.get_mutations():
-                alt_name = rename(alt_name)
-                # this node must run after the prior writer
-                add_user(alt_name, node)
-                node.add_fake_dep(StarDep(alt_name, mode=node_mode))
-                for other_node in name_to_users[alt_name].items:
-                    # this node must run after all prior readers
-                    other_name = rename(other_node.get_name())
-                    known_dep_node_names = dep_closure(node.get_name())
-                    if other_name not in known_dep_node_names:
-                        # If this node already directly or indirectly depends on other_node,
-                        # we don't need to insert an extra dep.
-                        node.add_fake_dep(WeakDep(other_name))
-                        add_user(other_name, node, is_weak=True)
+            # Handle output mutations
+            for buf in node.get_outputs():
+                # a node will mutate either 0 or 1 buffers
+                assert len(buf.get_mutations()) <= 1
+                for alt_name in buf.get_mutations():
+                    alt_name = rename(alt_name)
+                    # this node must run after the prior writer
+                    add_user(alt_name, node)
+                    node.add_fake_dep(StarDep(alt_name, mode=node_mode))
+                    known_dep_node_names = dep_closure(node)
+                    for user in name_to_users[alt_name].items:
+                        assert isinstance(user.node, BaseSchedulerNode)
+                        for other_name in user.node.get_buffer_names():
+                            # this node must run after all prior readers
+                            other_name = rename(other_name)
+                            if other_name not in known_dep_node_names:
+                                # If this node already directly or indirectly depends on other_node,
+                                # we don't need to insert an extra dep.
+                                node.add_fake_dep(WeakDep(other_name))
+                                add_user(other_name, node, is_weak=True)
 
             # add normal non-mutation dependencies
             for read in node.read_writes.reads:
-                is_weak = isinstance(read, WeakDep)
-                add_user(read.name, node, node.can_inplace(read), is_weak)
+                if not isinstance(read, WeakDep):
+                    add_user(read.name, node, node.can_inplace(read))
```

Comment:
what happened here?

Reply:
When we add the `WeakDep` above we call `add_user` at the same time, so this was redundant.